src/cpp/session/SessionSourceDatabase.cpp

```diff
@@ -242,6 +243,37 @@ Error attemptContentsMigration(json::Object& propertiesJson,
    return writeStringToFile(contentsPath, contents);
 }
 
+bool isIntendedAsReadOnly(const std::string& contents,
+                          std::string* pAlternative)
+{
+   boost::smatch match;
+   
+   // check for roxygen token specifically
+   boost::regex reRoxygen(
+            "% Generated by roxygen2: do not edit by hand\n"
+            "% Please edit documentation in ([^\n]+)");
```

Comment:
Does this need to account for files that use other comment delimiters (e.g. `#` for `NAMESPACE`)?

Reply:
In this case no, since `NAMESPACE` files don't have a single generator that we can link the user back to. (The default check below that checks for the text 'do not edit by hand' will still catch these cases)

Reply:
Okay, sounds good. I think the roxygen2 check here might be a little too sensitive to roxygen2 changes, but I can't think of a way we can do better without substantially more plumbing (e.g. providing some mechanism for the package to give us the string and/or mining the package itself).

Reply:
I'll try and make sure we don't change accidentally — https://github.com/klutometis/roxygen/issues/709